### ISSUE TITLE
Multifox doesn't work with Multi-Process

### ISSUE THREAD

**Issue by Sekuiya** [hultmann/multifox#215]
So, yesterday Firefox Developer Edition gave me the option to activate this new feature called Multi-Process (you can read more about it here: https://wiki.mozilla.org/Electrolysis ).
Upon activating, I noticed that the profiles stopped working.

Simply enough, I have two tabs, with different profiles, different Facebook accounts. Since I activated that feature, even though Multifox clearly shows that in one he is on the Default Profile and the other is in another profile, which for the sake of conversation I'll call it Profile 1, the same account appears on both tabs. Even if I log out in Profile 1 and log in in the other account, the moment I refresh the Default Profile it appears like they're the same profile and the other account appears logged there.

Deactivating Multi-Process makes Multifox work again.

Version used: Multifox 3.2.1 with Firefox Developer Edition 40.0a2 (2015-05-16)

**Comment by hultmann** [comment]
See also https://bugzilla.mozilla.org/show_bug.cgi?id=1047432

**Comment by art-solopov** [comment]
I confirm this. Note that you _can_ launch a e10s-free window in the Dev Edition. Just look for the corresponding menu item in the menu configuration screen.

**Comment by andresilva** [comment]
Is there any progress / roadmap towards fixing this?

**Comment by hultmann** [comment]
Hopefully before 2016.

My current plan is to investigate docShell.setIsApp() method, I could get rid of a bunch of code. There are advantages (new features) and disadvantages (new bugs) to this approach.

If it doesn't work then it will probably mean the end of Multifox, because fixing it is not worth the time it would take.

**Comment by andresilva** [comment]
Great! I don't have any experience with add-on development so I'm afraid I can't help. But I'm glad that you have a plan since this extension is really useful and important. Cheers.

**Comment by hultmann** [comment]
Mozilla just announced they will deprecate the XPCOM extension model and replace it with the Chrome model: https://blog.mozilla.org/addons/2015/08/21/the-future-of-developing-firefox-add-ons/

That means even when I fix the e10s thing, Multifox will eventually break again. This time it will probably be impossible to fix!

So the only sane thing to do is just stop spending time with Multifox. Sorry folks.

**Comment by k-bx** [comment]
Too bad to hear this. Firefox seems to have plans to support multiple identities "by design", hopefully they'll make it.

@hultmann huge thanks for MultiFox, it worked so well during a very long period of time and saved me from frustration a lot :)

**Comment by MurzNN** [comment]
@hultmann - huge thanks for developing Multifox from me too! It's too hard to hear that Multifox will not work on next Firefox versions :disappointed: 

Maybe new Firefox WebExtensions API have function alternatives from XPCOM to port Multifox code, or it is so limited, and make Multifox using WebExtensions API  is not possible?

**Comment by MurzNN** [comment]
Or maybe AllAccounts https://github.com/hultmann/allaccounts have use newer code, that will be easily port to new Firefox WebExtensions API?

**Comment by hultmann** [comment]
Currently WebExtensions support a subset of Chrome extension API. It is still not defined if something like Multifox/AllAccounts will be supported.

BTW, it seems Mozilla is planning a Multifox-like feature for Firefox.

**Comment by andresilva** [comment]
@hultmann can you link me to the issue about adding this feature to Firefox?

**Comment by hultmann** [comment]
yep: https://bugzilla.mozilla.org/show_bug.cgi?id=1191418

**Comment by ciuly** [comment]
For those not fully understanding mozillas bug/ticket system, is the Firefox new feature related to multifox functionality actually being done, or still in the design phase? That meta ticket depends on too many tickets for me (at least) to figure out if the necessaries for multifox-like browsing are made a reality or just planned to be sometime in the misty future.
Thanks.

**Comment by k-bx** [comment]
@ciuly if you mean feature which will provide similar functionality ("contextual identities") – it's under development, which seems active in recent few weeks

**Comment by ciuly** [comment]
Ok, so to make thing clear, all the tickets linked there in 1191418 as Depends On need to be resolved before "contextual identities" becomes a usable thing?

**Comment by k-bx** [comment]
Don't know, I would assume if some are not essential enabling this feature might start working earlier via some configuration option.

**Comment by hultmann** [comment]
For those interested, Brave (a Chromium-based web browser) has Multifox-like functionality built-in (Menu - New Session Tab).

**Comment by dnviti** [comment]
So is multifox at it's end? Because i deeply need it for work and i would like to enable the multiprocess feature...

**Comment by MurzNN** [comment]
Seems that Firefox 50 will have multifox-like feature for switching profiles: http://www.ghacks.net/2016/06/15/firefox-container-tab/
Hope that they release this feature before broke Multifox compatibility!

**Comment by k-bx** [comment]
Yeah. Bad news is, seems like it'll work in a bit different manner, so I'm still not sure it'll cover same use-cases.

**Comment by MurzNN** [comment]
Let's hope that @hultmann can write new version of extension, that extend functionality of Firefox Container tab for same use-cases! ;-)

**Comment by hultmann** [comment]
Sorry folks, I barely use Firefox nowadays. :P

**Comment by MurzNN** [comment]
Which browser did you use at now? Internet Explorer? :-) I see only one browser that I can customize for my needs - Firefox, with no alternatives...

**Comment by hultmann** [comment]
I'm working on my own custom browser using Electron. Unbeatable customization. :-)

**Comment by dnviti** [comment]
@hultmann We do not need a "Custom Browser". There are plenty of that on the internet. We need a standard to work with and need that extension. I moved to Chrome in the meantime: I've found a good extension that behaves like this (and better, saving sessions on cloud). Plus this extension brakes some functions of the Firefox UI (such as bookmarks editing popup) so, nowadays, it is totally useless.

**Comment by MurzNN** [comment]
@dnviti, can you provide link to extension? Chrome have very bad interface customization ability (for example, no tree style tabs, or vertical tabs), so it not useful for me, but seems that I must use it for switch accounts like Multifox :(

**Comment by MurzNN** [comment]
And seems that Container Tab functional in Firefox 53 alpha works much better, so I hope that they repeats Multifox functional in Firefox core. Here is main discussion about this feature: https://bugzilla.mozilla.org/show_bug.cgi?id=1191418
And wiki: https://wiki.mozilla.org/Security/Contextual_Identity_Project

**Comment by dnviti** [comment]
This is it: https://sessionbox.io/discover
You have also the cloud feature that allows you to store any number of sessions and then reuse them on-the-go.
They also quickly resolved an issue i reported about losing session on popup open, so now popup also maintain the parent session.
That is a great extension, i use it for work and it is perfect.

**Comment by MurzNN** [comment]
sessionbox.io releases Firefox version of extension after my feature request, so at now we can use it.

Also Firefox developers tells that  they expose ContextualIdentityService and containers as WebExtension APIs and we would do extensions for change tab profile with actions like this:
1. a context menu entries for changing container.
2. take the URL of the selected tab
3. open a new tab in the same position of the old one, using the new container ID.
4. close the previous tab.

@hultmann, will you can write easy extension as alternative to Multifox in future, when WebExtension API will be available?

**Comment by hultmann** [comment]
@MurzNN nope :)

**Comment by endolith** [comment]
Waterfox will continue to support these extensions after Mozilla drops them, can you continue developing it for people who use that?
